### PR TITLE
Infra/picker new file

### PR DESCRIPTION
**Pull request by ethanshar**
## Description
Remove old implementation of Picker. 
Fix issues with Picker Dialog when passing useNativePicker - related to issues with extracting dialog props 

## Changelog
Fix error with Picker dialog when passing useNativePicker

### CODE REVIEW

File: src/components/picker/NativePicker.js [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
-import {BaseComponent} from '../../commons';
+import React, {Component} from 'react';
 import TextField from '../textField';
 import PickerDialog from './PickerDialog';
 import TouchableOpacity from '../touchableOpacity';
 import {Colors} from '../../style';
 import {WheelPicker} from '../../incubator';
 
-class NativePicker extends BaseComponent {
-  static displayName = 'IGNORE';
```
Comment (Inbal-Tish) [inline src/components/picker/NativePicker.js:11]: Should we add the displayName?

File: src/components/picker/PickerDialog.js [hunk 2]
```diff
@@ -1,17 +1,14 @@
...
 
-import {BaseComponent} from '../../commons';
-import {extractComponentProps} from '../../commons/modifiers';
 import Dialog from '../dialog';
 import View from '../view';
 import Text from '../text';
 import {Colors} from '../../style';
 
-class PickerDialog extends BaseComponent {
-  static displayName = 'IGNORE';
```
Comment (Inbal-Tish) [inline src/components/picker/PickerDialog.js:14]: Also here, don't we need the displayName static?

**Comment by Inbal-Tish** [comment]
@ethanshar Looks good, just a litle question regarding the DisplayName static and the picker tests are failing...

File: src/components/picker/NativePicker.js (hunk 1 above)
Comment (ethanshar) [reply src/components/picker/NativePicker.js:11]: Fixed. 
It helps when errors prints callstack

File: src/components/picker/PickerDialog.js (hunk 2 above)
Comment (ethanshar) [reply src/components/picker/PickerDialog.js:14]: Fixed

**Comment by ethanshar** [comment]
> @ethanshar Looks good, just a litle question regarding the DisplayName static and the picker tests are failing...

Fix tests, I moved to render tests instead,

File: src/components/picker/PickerDialog.js [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
+class PickerDialog extends Component {
+  static displayName = 'NativePicker';
```
Comment (Inbal-Tish) [inline src/components/picker/PickerDialog.js:12]: You have 'NativePicker' already in the NativePicker component. This should be the 'PickerDialog' and the android one should be 'PickerDialog.android', no?

Comment (ethanshar) [reply src/components/picker/PickerDialog.js]: Oops, bad copy-paste. 
And no, both should be `PickerDialog`, they can't co-exist. One is for iOS and the other is for Android

**Review by Inbal-Tish — APPROVED** [review]

**Pull request merged**